src/Game/Managers/AudioManager.cs

```diff
@@ -148,6 +163,25 @@ public void UpdateCurrentMusicVolume()
             }
         }
 
+        public void UpdateCurrentSoundsVolume()
+        {
+            if (!_canReproduceAudio)
+                return;
+
+            if (Engine.Profile == null || Engine.Profile.Current == null || !Engine.Profile.Current.EnableSound)
+                return;
+
+            float volume = Engine.Profile.Current.SoundVolume / Constants.SOUND_DELTA;
+
+            if (volume < -1 || volume > 1f)
+                return;
+
+            foreach (UOSound sound in _currentSounds.ToArray())
```

Comment:
Why .ToArray() ?

Reply:
I added that so I could remove from within the loop. But you are right, I will follow your for suggestion.